torch/testing/_creation.py

```diff
@@ -153,8 +154,10 @@ def clamp(a, l, h):
                         " To request support, file an issue at: https://github.com/pytorch/pytorch/issues")
 
     if noncontiguous and result.numel() > 1:
-        result = torch.repeat_interleave(result, 2, dim=-1)
-        result = result[..., ::2]
+        # result = torch.repeat_interleave(result, 2, dim=-1)
```

Comment:
Original implementation of noncontiguous hides some stride bugs. 

TensorIterator treats "densely packed but non-contiguous" (e.g. `t=torch.rand(2,2).t()`) differently from "non-densely packed" tensor (e.g. ` t with size [2,2] and stride [4,2] `).

Reply:
Also, we need to mix contiguous and noncontiguous tensors in the input tensors... 
I am using random.random() < 0.5 as a hack to randomly generate noncontiguous tensors.

Reply:
The "proper" way to do this, I suppose, would be to add more parameters (noncontiguous/nonoverlapping) for the various parameters, and then have the call sites iterate through the combinations of options, but I'm not sure how to do this without blowing up test time too.

Reply:
I don't think we should change how `make_tensor` generates noncontiguous tensors by default. We could add new tensor generation strategies to `make_tensor`, however. In many cases this proposed change wouldn't even generate a noncontiguous tensor?

Reply:
Although we could look at aligning `make_tensor`'s noncontiguity option with the `noncontiguous_like` helper if we think it's better:

https://github.com/pytorch/pytorch/blob/874625e039274b83e70163733f6f2c1689b0de4e/torch/testing/_internal/common_utils.py#L2934

Reply:
I think the right way to go about this would be to assert that all opnifos have both permuted and non-dense, strided inputs (see https://github.com/pytorch/pytorch/issues/86381). And to add an arg like [here](https://github.com/pytorch/pytorch/pull/85976).

Reply:
Ideally we would generate the permuted and non-dense inputs from a simpler specification so we don't revisit PyTorch's historic test creep, where adding an operator required adding an impossible number of tests to validate all its behavior. Maybe an extension of a test like `test_noncontiguous`:

https://github.com/pytorch/pytorch/blob/85a79a7f506dadbc269cf2abb7536f64ab49074d/test/test_ops.py#L487

@ezyang is correct that we need to be concerned about test time, too, of course, but we could run such a test rarely (or even require it be run manually) if it took too much time.